src/components/FullscreenLoadingIndicator.tsx

```diff
@@ -15,22 +19,69 @@ type FullScreenLoadingIndicatorProps = {
     /** Size of the icon */
     iconSize?: FullScreenLoadingIndicatorIconSize;
 
+    /** Timeout for the activity indicator after which we fire a log about abnormally long loading */
+    timeout?: number;
+
     /** The ID of the test to be used for testing */
     testID?: string;
 
     /** Extra loading context to be passed to the logAppStateOnLongLoading function */
     extraLoadingContext?: ExtraLoadingContext;
 };
 
-function FullScreenLoadingIndicator({style, iconSize = CONST.ACTIVITY_INDICATOR_SIZE.LARGE, testID = '', extraLoadingContext}: FullScreenLoadingIndicatorProps) {
+function FullScreenLoadingIndicator({
+    style,
+    iconSize = CONST.ACTIVITY_INDICATOR_SIZE.LARGE,
+    timeout = CONST.TIMING.ACTIVITY_INDICATOR_TIMEOUT,
+    testID = '',
+    extraLoadingContext,
+}: FullScreenLoadingIndicatorProps) {
     const styles = useThemeStyles();
+    const [showGoBackButton, setShowGoBackButton] = useState(false);
+    const [translateY, setTranslateY] = useState(0);
+    const footerRef = useRef<View>(null);
+
+    const {translate} = useLocalize();
+
+    useLayoutEffect(() => {
+        if (!showGoBackButton || !footerRef.current) {
+            return;
+        }
+        footerRef.current.measure((x, y, width, height) => {
+            setTranslateY(height / 2);
+        });
+    }, [showGoBackButton]);
```

Comment:
How about mounting the back button at the beginning and changing its opacity after 10 seconds? That way we could get rid of the layout effect and measuring